Makefile

```diff
@@ -191,3 +176,24 @@ endef
 define local_image_exists
 $(shell docker images -q $1 2> /dev/null)
 endef
+
+# arguments:
+# $1 - release image repo, e.g. mesosphere/spark
+# $2 - release image tag, e.g spark-2.4.3-hadoop-2.9-k8s
+# $3 - dev image full name, e.g mesosphere/spark-dev:ab36f1f3691a8be2050f3acb559c34e3e8e5d66e
+define tag_and_push_image
+	$(eval RELEASE_IMAGE_FULL_NAME=$(1):$(2))
+	# check, if specified image already exists to prevent overwrites
+	if [[ -z "$(call remote_image_exists,$(1),$(2))" ]]; then
+		# check, if dev image is present in local docker registry
+		[[ -z "$(call local_image_exists,$(3))" ]] && docker pull $(3)
```

Comment:
does it make sense to do `docker pull` without checking for image presence? the time penalty is minimal if image is already present locally

Reply:
yes, you are right, this check is unnecessary.